### PR TITLE
refactor(dh): use Signals in components

### PR DESCRIPTION
**Pull request by dzhavat**
## References

- #0000

### CODE REVIEW

File: libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts [hunk 1]
```diff
@@ -101,15 +107,23 @@ export class DhDelegationTabComponent {
...
+    });
 
-    this.statusControl.valueChanges.subscribe((value) => {
-      this.filterDelegations(value);
+    effect(() => this.delegationsByType.set(dhGroupDelegations(this.delegationsRaw())), {
+      allowSignalWrites: true,
     });
+
+    this.statusControl.valueChanges
+      .pipe(takeUntilDestroyed(this.destroyRef))
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts:121]: Suggestion: No need for a destoryRef inside a constructor

File: libs/dh/market-participant/actors/feature-delegation/src/lib/overview/dh-delegations-overview.component.ts [hunk 2]
```diff
@@ -65,20 +50,19 @@ import { DhDelegationTableComponent } from '../table/dh-delegation-table.compone
...
 })
 export class DhDelegationsOverviewComponent {
-  private readonly changeDetectorRef = inject(ChangeDetectorRef);
   private readonly permissionService = inject(PermissionService);
 
   delegationsByType = input.required<DhDelegationsByType>();
 
-  canManageDelegations$ = this.permissionService
-    .hasPermission('delegation:manage')
-    .pipe(tap(() => this.changeDetectorRef.detectChanges()));
```
Comment (mimse) [inline libs/dh/market-participant/actors/feature-delegation/src/lib/overview/dh-delegations-overview.component.ts:83]: Praise: Omg yes!

**Review by mimse — CHANGES_REQUESTED** [review]
Praise: Beautiful 💯

File: libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts (hunk 1 above)
Comment (dzhavat) [reply libs/dh/market-participant/actors/feature-delegation/src/lib/dh-delegation-tab.component.ts]: Fixed

**Review by mimse — APPROVED** [review]

**Pull request merged**